rfc/0006-kubewarden-versioning.md

```diff
@@ -0,0 +1,226 @@
+
+Hence we will perform a minor update.
+
+We are going to tag `v0.6.0` of all these components and release them:
+
+* kubewarden controller
+* policy server
+* kwctl
+
+This will happen **regardless** of the changes done to these projects. For example, kwctl is going to have a new release `v0.6.0` even if it didn't have any new commit since `v0.5.3`.
+
+The helm chart is updated in this way:
+
+* The version of container images of policy server and kubewarden
+  controller are upgraded to be `v0.6.0`
+* It's `version` attribute is set to `v0.10.4` because no significant changed happened
+* The `appVersion` attribute is set to `0.6.0`
+
+### A patch is done to kwctl
+
+A series of minor issues are found inside of kwctl. We decide to
+tag a new release that includes these fixes.
+
+The version of `kwctl` is updated from `0.6.0` to `0.6.1`.
+No other releases are done
+
+### A patch is done to kubewarden controller
+
+A minor issue is fixed with kubewarden controller. We decide to
+tag a new release that includes the fix.
+
+The following actions are done:
+
+* We tag a new release of kubewarden-controller: `0.6.1`
+* We update the helm chart that makes use of this image:
+  * We update the references of the image to ensure `0.6.1` is being pulled
+  * We update the `version` number of the chart itself to be
+    change from `0.10.4` to `0.10.5`.
+    
+    **Note:** we don't change the `appVersion` attribute
+
+### A major security issue is found inside of policy server
+
+A major security issue is found inside of policy server. Changes are done to both policy server and kubewarden controller.
+
+We decide to release a new version of the whole stack, because of
+the size of the changes and the impact. We will bump the
+kubewarden stack from `0.6.0` to `0.7.0`.
+
+
+We proceed with the following actions:
+
+* Tag a new release of kubewarden controller: `0.7.0`
+* Tag a new release of policy server: `0.7.0`
+* Tag a new release of kwctl: `0.7.0`. We do that even though
+  no actual change has been done to kwctl
+* helm chart:
+  * Update all the references to the kubewarden controller and
+    policy server images
+  * Update `version`: this time the changes involve some updates to
+    the helm chart itself. We deem this is worth a minor update.
+    Because of that, the `version` goes from `0.10.5` to `0.11.0`
+  * Update the `appVersion` to be `0.7.0`
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- There are times when the version of one of the components of
+  the "public stack" is bumped even if no actual changes have
+  been done to its source code.
+- The global stack version does not make use of the `.patch`
+attribute of Semantic Versioning. That's because we want to
+leave some freedom to each component of the "public stack".
+
+# Alternatives
+[alternatives]: #alternatives
+
+[This](https://github.com/kubewarden/kubewarden-controller/pull/182)
+rejected proposal.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
```

Comment:
How to execute the release process? As we have a bunch o different versions the release process can be very error prone. Therefore, I believe we should try to automate or, at least, document **very** well how to do it.

Reply:
I would prefer to automate the whole process

Reply:
We already have automation at the single project level, we would need something that orchestrates that.

I would however wait a little before investing time into doing that to understand:
  * If this versioning proposal sound, are we going to face unexpected issues?
  * How often does that release happen? What is the tradeoff between investing time into automating the "high level orchestrator" vs following a clearly written playbook?

Reply:
I kept track of this conversation inside of the RFC